### ISSUE TITLE
factor: a standard working directory is set to be in /nix/store

### ISSUE THREAD

**Issue by eighal** [NixOS/nixpkgs#46300]
## Issue description

Standard use of factor (including following the second step in the tutorial) requires using working directories.  The default location of the first working directory a new user comes across is set by this package to be in /nix/store.

That means the working directory is read-only, of course, and so it can't be used for its intended purpose.

There are work-arounds, but they don't go well with standard practice (I think) or the tutorial (definitely).

### Steps to reproduce

Install the factor package.

Run the first two steps of the tutorial:

```
USE: tools.scaffold

"palindrome" scaffold-work
```

Error message:

```
Unix system call “mkdir” failed:

Read-only file system (30)

It was called with the following arguments:

"/nix/store/ny5pmkk5hghbs6gmd22j95zkx8nxw7vd-factor-lang-0.97/..."
511
```

## Technical details

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the
results.

```
 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.68, NixOS, 19.03pre151837.ca2ba44cab4 (Koi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.1`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

```

**Comment by eighal** [comment]
@vrthra: This needs someone who knows a bit about setting defaults in factor's ecosystem.  I hope to learn about it one day, but at the moment I have no idea at all.  Any chance you can look at this?  Thanks very much.

**Comment by spacefrogg** [comment]
This was fixed in #48727.
